From ecadfad5426aec73d2c7e85cd859b23f538b3d61 Mon Sep 17 00:00:00 2001
From: Giulio Fidente <gfidente@redhat.com>
Date: Thu, 25 Jun 2015 03:22:15 +0200
Subject: [PATCH] Use mode tcp for glance-registry balancing

The glance-registry service is returning 401 to httpchk, which
makes haproxy think it is down. This change switches the check
mode to tcp.

Closes-Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1234637
Closes-Bug: 1468566

Change-Id: Icdd80aa9cd56e5afd3707eb7fa38aaedb8535af6
---
 tripleo/manifests/loadbalancer.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tripleo/manifests/loadbalancer.pp b/tripleo/manifests/loadbalancer.pp
index ac001fd..102deeb 100644
--- a/tripleo/manifests/loadbalancer.pp
+++ b/tripleo/manifests/loadbalancer.pp
@@ -674,7 +674,7 @@ class tripleo::loadbalancer (
       ipaddress        => hiera('glance_registry_vip', $controller_virtual_ip),
       ports            => 9191,
       options          => {
-        'option' => [ 'httpchk GET /' ],
+        'mode' => 'tcp',
       },
       collect_exported => false,
     }
